Feature 337 xml spec doc

bikegeek opened the pull request:
## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

- verified tests still pass for test_xml.py

- verified that updates to the XML schema (to accommodate listing data files individually) successfully load the listed files (via METviewer docker)
- verified GHA tests all pass
- visually checked the content on ReadTheDocs


- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- read the User's Guide section, section 4 Background 
   -  verify that content is organized and instructions for setting up XML specification files is correct
   

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [x] Do these changes include sufficient testing updates? **[Yes]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Do these changes introduce new SonarQube findings? **[No]**</br>
If **yes**, please describe:

- [x] Please complete this pull request review by **[for RC1 release]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/appendixA.html#metplus-components-python-packages) table.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issue
Select: **Milestone** as the version that will include these changes
Select: **Coordinated METplus-X.Y Support** project for bugfix releases or **METdataio-X.Y.Z Development** project for official releases
- [x] After submitting the PR, select the :gear: icon in the **Development** section of the right hand sidebar. Search for the issue that this PR will close and select it, if it is not already selected.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

bikegeek committed:
Delete METreformat/test/eclv_mpr_pct_vl1l2_vcnt.xml

no longer used

bikegeek committed:
Delete METreformat/test/ens_stat_ecnt.xml

no longer used

bikegeek committed:
Delete METreformat/test/mctc_mcts.xml

no longer used

bikegeek committed:
Delete METreformat/test/point_stat.xml

no longer used

bikegeek committed:
Initial updates for detailed information on loading data.

bikegeek committed:
Initial version-needs cleaning up

bikegeek committed:
Merge branch 'feature_337_xml_spec_doc' of https://github.com/dtcenter/METdataio into feature_337_xml_spec_doc

bikegeek committed:
increas numbered from 4 to 6

bikegeek committed:
Delete docs/Users_Guide/common.rst

outdated information, replacing XML specification information in the load_data.rst file

bikegeek committed:
remove any links to common.rst

bikegeek committed:
Merge branch 'feature_337_xml_spec_doc' of https://github.com/dtcenter/METdataio into feature_337_xml_spec_doc

bikegeek committed:
Work in progress, need to add more content

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METdataio into feature_337_xml_spec_doc

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METdataio into feature_337_xml_spec_doc

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METdataio into feature_337_xml_spec_doc

bikegeek committed:
Add example for line_type element

bikegeek committed:
Add checking for the load_files element to the schema

bikegeek committed:
Configs for loading specific files

bikegeek committed:
Merge branch 'develop' of https://github.com/dtcenter/METdataio into feature_337_xml_spec_doc

bikegeek committed:
additional updates, more are pending.  Mostly format changes.

bikegeek committed:
Removed unused content for folder_tmpl information.

bikegeek committed:
folder_tmpl min occurrence changed to 0 because of load_files support

bikegeek committed:
Accidentally removed the <verbose> element.  Put this back in.

bikegeek committed:
Added description of flags.

bikegeek committed:
add the valid_times in the folder_tmpl to be consistent with the load_val elements in the exmample

bikegeek committed:
Changed order of elements for consistency

bikegeek committed:
Added content for the load_val/field for config

bikegeek committed:
Finished content for folder_tmpl

bikegeek committed:
Completed instructions/descriptions for generating the XML specification file

bikegeek committed:
all content added for Background, Create the XMO Specification File, and Troubleshooting sections

bikegeek committed:
Added instructions for using mv_mysql.sql

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+.. dropdown:: Create MET linetype tables
 
-In the METdataio/METdbLoad/sql/scripts directory, there are two configuration files:
+ mysql -udbuser -pdbpasswd dbtable_name</path-to-METdataio/METdbLoad/sql/mv_mysql.sql
```
```suggestion
 mysql -udbuser -pdbpasswd dbtable_name </path-to-METdataio/METdbLoad/sql/mv_mysql.sql>
```
Please verify, but I think there needs to be a space after dbtable_name and an ending ">".  If that is correct, you should be able to just commit this suggested change.

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
 
-  * data_loading_config.yaml
+    - Replace *dbpasswd* with the password
 
```
```suggestion

- Replace *dbtable_name* with the database table name
```
Again, not sure if this is correct/needed or not, so please verify.

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+.. dropdown:: Create MET linetype tables
 
-In the METdataio/METdbLoad/sql/scripts directory, there are two configuration files:
+ mysql -udbuser -pdbpasswd dbtable_name</path-to-METdataio/METdbLoad/sql/mv_mysql.sql
```
```suggestion
 mysql -u<dbuser> -p<dbpasswd> <dbtable_name> </path-to-METdataio/METdbLoad/sql/mv_mysql.sql>
```
I added "<>" for all values that need replacing and will suggest updates to the text below.  I thought it might be more clear.

bikegeek replied:
This is a mysql command, the < is a redirect.

jprestop replied:
Phew! I'm glad you're paying attention.  Please pardon my ignornace!

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+ mysql -udbuser -pdbpasswd dbtable_name</path-to-METdataio/METdbLoad/sql/mv_mysql.sql
 
-  * db_load_specification.xml
+    - Replace *dbuser* with the username
```
```suggestion
    - Replace *<dbuser>* with the username
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+    - Replace *dbuser* with the username
 
-  * data_loading_config.yaml
+    - Replace *dbpasswd* with the password
```
```suggestion
    - Replace *<dbpasswd>* with the password
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
-is a template YAML configuration file.  The *data_loading_config.yaml* file contains
-information about the database (username, password, database name, etc.). This information is used by the
-*generate_xml_spec.py* script to generate the XML specification file which is then used to load data into the database.
+    - Replace *path-to-METdataio* with the path where the METdataio source code is saved
```
```suggestion
    - Replace *<dbtable_name>* with the database table name     

    - Replace *<path-to-METdataio>* with the path where the METdataio source code is saved
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+The elements in the XML specification file **must** adhere to the **order specified** by the
+XML schema and conform to size and number of element limitations.
+
+.. dropdown:: This is an **example XML specification file** that is **valid**:
```
```suggestion
.. dropdown:: An **example XML specification file** that is **valid**
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+
+    .. literalinclude:: ../../METdbLoad/test/Examples/example_load_specification.xml
+
+.. dropdown:: This is another **example XML specification file** that is **valid**:
```
```suggestion
.. dropdown:: Another **example XML specification file** that is **valid**
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+
+
+
+.. dropdown:: This is the **XML schema** that is used to **validate the XML specification file**:
```
```suggestion
.. dropdown:: The **XML schema** that is used to **validate the XML specification file**
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
-Replace the *path-to-your-dir* with the actual path to where this file is to be saved.
 
-Change directory to the location where the *data_loading_config.yaml* file was copied.
+- $METDATAIO is the path to the location of the cloned or forked METdataio source code.
```
```suggestion
- $METDATAIO_HOME is the path to the location of the cloned or forked METdataio source code.
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+   cd path-to-your-dir/load_specification.xml
 
-  * username
+- replace *path-to-your-dir* with the full path where you are saving the XML specification file
```
```suggestion
- Replace *path-to-your-dir* with the full path where the XML specification file will be saved.
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+- $METDATAIO is the path to the location of the cloned or forked METdataio source code.
 
-Open the data_loading_config.yaml file:
+- Replace the *path-to-your-dir* with the actual path to where this file is to be saved.
```
```suggestion
- Replace the *path-to-your-dir* with the actual path to where this file will be saved.
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+payloads that can compromise the loading of data.
+The elements in the XML specification file **must** adhere to the **order specified** by the
+XML schema and conform to size and number of element limitations.
+
```
```suggestion
Click to expand the example XML specifications and the XML validation schema below.

```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+expand). Remember to update the username and password that is applicable to your database.
 
-  * host
+.. dropdown:: The following is an explanation of the required and optional elements and any limitations
```
```suggestion
.. dropdown:: An explanation of the required and optional elements and any limitations
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
-  * port
+  *These are element names. The XML angle brackets (<>) as seen in the XML specification file are omitted for simplicity*
 
+  **!!!NOTE!!!**
```
```suggestion
  ..note::
```
See "Notes" in our documentation for more info: https://metplus.readthedocs.io/en/develop/Contributors_Guide/documentation.html#notes
It seems like a good place to use it here.

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+  **!!!NOTE!!!**
 
-Update the path to the schema location, provide the full path to the *mv_sql_mysql.sql* schema file:
+  The **order of the elements** in the XML specification file is crucial. **DO NOT** modify the order of the following elements.
```
```suggestion
    
    The **order of the elements** in the XML specification file is crucial. **DO NOT** modify the order of the following elements.
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
 
-Update the path to the schema location, provide the full path to the *mv_sql_mysql.sql* schema file:
+  The **order of the elements** in the XML specification file is crucial. **DO NOT** modify the order of the following elements.
+  Indentation is used to indicate hierarchical relationships between elements.
```
```suggestion
    Indentation is used to indicate hierarchical relationships between elements.
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+      - container for other elements that define connection information, flags, data input, etc.
 
-  * xml_specification
+    *The following elements pertain to logging into the database*
```
```suggestion
    *The following elements pertain to logging into the database.*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+                  - METdataio sets default to False if this element is absent
 
-  python met_db_load.py /path-to/load_met.xml
+    *The following elements are used to define the format of multiple input data directories that are (optionally) organized by datetime*
```
```suggestion
    *The following elements are used to define the format of multiple input data directories that are (optionally) organized by datetime.*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+       - indicate whether or not to force load paths/files that already exist
+       - True or False (case insensitive)
+
+    *The following elements indicate which group the database should be assigned and a description*
```
```suggestion
    *The following elements indicate which group the database should be assigned and a description.*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+
+    **Determine whether to use folder_tmpl or load_files to define where your input data resides (descriptions below):**
+
+    *The following defines the location of the input data to be loaded into the database based on data organized by datetime (and any other criteria)*
```
```suggestion
    *The following defines the location of the input data to be loaded into the database based on data organized by datetime (and any other criteria).*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+
+
+
+    *The following defines the location of specific input data files to be loaded into the database*
```
```suggestion
    *The following defines the location of specific input data files to be loaded into the database.*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+
+                </load_files>
+
+    *The following describe the linetypes to load*
```
```suggestion
    *The following describe the linetypes to load.*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+
+                   </line_type>
+
+    *The following option allows one to indicate whether to save the XML commands into the database*
```
```suggestion
    *The following option allows one to indicate whether to save the XML commands into the database.*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,694 @@
+         <load_xml>true</load_xml>
+
+
+    *The following allows one to create a note into the instance_info database table*
```
```suggestion
    *The following allows one to create a note into the instance_info database table.*
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -359,3 +721,16 @@ Troubleshooting
+  * - Solution:
+    - This error is typically encountered when one of the following conditions exist as a result of failing the XML validation step:
+
+         - the order of the elements in your XML specification file is inconsistent with the order expected
```
```suggestion
         - the order of the elements in the XML specification file is inconsistent with the order expected
```
We try to keep "you" language out of the guides.  Suggest changing to "the" unless you think it is unclear.

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -359,3 +721,16 @@ Troubleshooting
+    - This error is typically encountered when one of the following conditions exist as a result of failing the XML validation step:
+
+         - the order of the elements in your XML specification file is inconsistent with the order expected
+         - your XML specification file is missing one or more mandatory elements
```
```suggestion
         - the XML specification file is missing one or more mandatory elements
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -359,3 +721,16 @@ Troubleshooting
+         - the order of the elements in your XML specification file is inconsistent with the order expected
+         - your XML specification file is missing one or more mandatory elements
+         - one or more elements has exceeded size limits specified in the XML schema
+         - you have addition XML elements that are not expected
```
```suggestion
         - there are additional XML elements that are not expected
```

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -359,3 +721,16 @@ Troubleshooting
+         - one or more elements has exceeded size limits specified in the XML schema
+         - you have addition XML elements that are not expected
+
+       **Refer to the section **Create the XML Specification File** to verify that your XML specification file is correct.**
```
```suggestion
       **Refer to the section **Create the XML Specification File** to verify that the XML specification file is correct.**
```

jprestop commented:
Hi @bikegeek.  Wow!  This was a lot of work.  Thank you for all of your effort.

I have read the User's Guide section, section 4 Background.  I am comfortable verifying that the content is organized.  I made some small mostly nit-picky suggestions. Unfortunately, I don't feel comfortable reviewing/approving that the instructions for setting up XML specification files is correct.  Do you have someone else that you could ask to review that part who is more familiar with this?

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek commented:
For clarification, I added an explanation in one of the bullets and also
added spacing around the redirect.

On Wed, Nov 6, 2024 at 2:28 PM Julie Prestopnik ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In docs/Users_Guide/load_data.rst
> <https://github.com/dtcenter/METdataio/pull/351#discussion_r1831732291>:
>
> >
> -In the METdataio/METdbLoad/sql/scripts directory, there are two configuration files:
> + mysql -udbuser -pdbpasswd dbtable_name</path-to-METdataio/METdbLoad/sql/mv_mysql.sql
>
> Phew! I'm glad you're paying attention. Please pardon my ignornace!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdataio/pull/351#discussion_r1831732291>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4UJHQPMQAPUPVLZOQFUT3Z7KCWJAVCNFSM6AAAAABRJRI2I6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMJZGQ2TQNRSHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

bikegeek committed:
Fixed some formatting/reorganizing and typos

bikegeek committed:
Replacing <bold>NOTE<bold> with RST syntax

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -2,332 +2,696 @@
+.. dropdown:: Create MET linetype tables
 
-In the METdataio/METdbLoad/sql/scripts directory, there are two configuration files:
+ mysql -udbuser -pdbpasswd dbtable_name</path-to-METdataio/METdbLoad/sql/mv_mysql.sql
```
@bikegeek Do we also need a line for " - Replace *dbtable_name* with the database table name"?

bikegeek replied:
added the following bullet:
Replace *dbtable_name* with the appropriate database name that begins with **mv** (e.g. mv_xyz )

to reinforce the mv_ prefix to table names

jprestop commented:
@bikegeek What do you think about changing the title from "Background" to "Using METdbLoad"?  I feel like "Background" doesn't cover the scope of the documentation well enough.

Also if you agree, please consider [updating this section](https://metdataio.readthedocs.io/en/feature_337_xml_spec_doc/Users_Guide/installation.html#id1) as well.
Perhaps this [section](https://metdataio.readthedocs.io/en/feature_337_xml_spec_doc/Users_Guide/installation.html#download) could say something like "Download the latest release" (and include a link to the releases page) or clone the METdataio repository with only "git clone https://github.com/dtcenter/METdataio" in the code block.
And this [section](https://metdataio.readthedocs.io/en/feature_337_xml_spec_doc/Users_Guide/installation.html#create-database) with these instructions:
```
cd [install]/metviewer
mysql -u[db_username] -p[db_password] -e'create database [db_name];'
mysql -u[db_username] -p[db_password] [db_name] < sql/mv_mysql.sql
```
could be moved to the [new page](https://metdataio.readthedocs.io/en/feature_337_xml_spec_doc/Users_Guide/load_data.html#background) as an example under "Before using the METdbLoad module, the database must exist and have the proper permissions (i.e. grant privileges to insert, delete, update, and index)."  And then the [Create Database](https://metdataio.readthedocs.io/en/feature_337_xml_spec_doc/Users_Guide/installation.html#create-database) section could just be removed or could point to the new, potentially renamed, "Using METdbLoad" page.

bikegeek committed:
Clean up, move some instructions to the Using METdbLoad secton

bikegeek committed:
Add a Create Database and Create an XML Load specification Document.  Clean up some formatting, migrated some content from the Insatllation section to this section.

bikegeek commented:
- Moved content related to database creation from the Installation section to the Using METdbLoad section
- Cleaned up some wording in the METdbLoad section

bikegeek committed:
Fixed formatting, comments that were in the code block are now outside the code block

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -1,333 +1,725 @@
-*generate_xml_spec.py* script to generate the XML specification file which is then used to load data into the database.
 
-Generate the XML Specification File
+    - Replace *path-to-METdataio* with the path where the METdataio source code is saved
```
```suggestion
    - Replace *dbuser* with the username
    - Replace *dbpasswd* with the password
    - Replace *dbtable_name* with the appropriate database name that begins with **mv** (e.g. mv_xyz )
    - Replace *path-to-METdataio* with the path where the METdataio source code is saved
```
It looks like we lost the other entries that we had previously, I think.  Please correct me if I am wrong.

@jprestop I see the suggested text, are you looking at a stale version of the  documentation: https://metdataio.readthedocs.io/en/feature_337_xml_spec_doc/Users_Guide/load_data.html???

jprestop replied:
@bikegeek this is in the Create MET linetype tables section.

jprestop replied:
@bikegeek You should be able to click "Commit suggestion" here if you agree.

jprestop commented on docs/Users_Guide/load_data.rst:
```diff
@@ -1,333 +1,725 @@
-Replace the *path-to-your-dir* with the actual path to where this file is to be saved.
 
-Change directory to the location where the *data_loading_config.yaml* file was copied.
+- $METDATAIO is the path to the location of the cloned or forked METdataio source code.
```
```suggestion
- $METDATAIO_HOME is the path to the location of the cloned or forked METdataio source code.
```
I believe this needs to be $METDATAIO_HOME and not $METDATAIO as the path references $METDATAIO_HOME.

jprestop replied:
@bikegeek You should be able to click "Commit suggestion" here if you agree.

bikegeek committed:
lost the last bullet point for the create database section on creating the database

bikegeek committed:
Remove last change, the text was correct

bikegeek commented:
I replaced what was originally there with this text, which was originally
under the Installation section.  Is that what you're wondering about?  I
also created a new section header:
Create an XML Load Specification Document

where instructions for the XML load specification file, so it is consistent
with the Create Database section.  Apologies for making further changes to
the documentation.




On Thu, Nov 7, 2024 at 3:51 PM Julie Prestopnik ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In docs/Users_Guide/load_data.rst
> <https://github.com/dtcenter/METdataio/pull/351#discussion_r1833470213>:
>
> >
> -Generate the XML Specification File
> +    - Replace *path-to-METdataio* with the path where the METdataio source code is saved
>
> ⬇️ Suggested change
>
> -    - Replace *path-to-METdataio* with the path where the METdataio source code is saved
> +    - Replace *dbuser* with the username
> +    - Replace *dbpasswd* with the password
> +    - Replace *dbtable_name* with the appropriate database name that begins with **mv** (e.g. mv_xyz )
> +    - Replace *path-to-METdataio* with the path where the METdataio source code is saved
>
> It looks like we lost the other entries that we had previously, I think.
> Please correct me if I am wrong.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METdataio/pull/351#pullrequestreview-2422316021>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA4UJHSCTWQE5BDAWOWZADTZ7PVGFAVCNFSM6AAAAABRJRI2I6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMRSGMYTMMBSGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jprestop commented:
> I replaced what was originally there with this text, which was originally
under the Installation section.  Is that what you're wondering about?  I
also created a new section header:
Create an XML Load Specification Document
where instructions for the XML load specification file, so it is consistent
with the Create Database section.  Apologies for making further changes to
the documentation.

I'm sorry, @bikegeek, I'm not sure what you are referring to in this comment with regard to "I replaced what was originally there with this text".  Where is "there"?

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

jprestop commented:
Hi @bikegeek. Please take a look at this comment with suggested changes. And, I’m not sure what you were referring to in this comment.

jprestop commented:
Hi @bikegeek. Please take a look at this [comment](https://github.com/dtcenter/METdataio/pull/351#discussion_r1833470213) with suggested changes.

I accepted the suggestion, is it not showing up?? I redid the accept and resolved the conversation.  


 And, I’m not sure what you were referring to in this [comment](https://github.com/dtcenter/METdataio/pull/351#issuecomment-2463419775).

I moved the database instructions from the Installation section to the using METdbLoad section, that is what I meant.

bikegeek committed:
Update docs/Users_Guide/load_data.rst

Co-authored-by: Julie Prestopnik <jpresto@ucar.edu>

bikegeek committed:
manually modify instructions to indicate path to location of cloned data

bikegeek committed:
Fixed instructions to use same language as earlier content for consistency

jprestop reviewed (approved):
@bikegeek Thanks for all of your work on this task and for iterating with me.  I know this was a huge effort.  I think it will be really helpful for users and appreciate all of your hard work and effort.  I approve this request.
